check_limits.py

```diff
@@ -1,36 +1,49 @@
-battery_limits = {"temperature": [0, 45], "soc": [20, 80], "charge_rate": [0, 0.8]}
+battery_limits = {'temperature': {'min' : 0, 'max' : 45}, 'soc': {'min' : 20, 'max' : 80}, 'charge_rate': {'min' : 0, 'max' : 0.8} }
 					
-warn_limit = 5
+def threshold_val(max):
+	warn_limit = 5
+	return (max * warn_limit) / 100
 
-lang_msg ={'DE' : {'warning' : "Warnung! Bitte halten Sie innerhalb des Bereichs",'Okay' : "Batterie ist in einwandfreiem Zustand", 'fail' : "Batterielimits A-Range!"},
-	   'EN' : {'warning' : "Warning! Please maintain wihtin Range", 'Okay' : "Battery in Perfect Condition", 'fail' : " Battery Limits out of Range!"}}
+lang_msg ={'DE' : {'warning' : "Warnung! Wert hat Schwellenwert erreicht" , 'Okay' : "Batterie ist in einwandfreiem Zustand", 'fail' : "Batterielimits A-Range!"},
+	   'EN' : {'warning' : "Warning! Value has reached threshold limit" , 'Okay' : "Battery in Perfect Condition", 'fail' : " Battery Limits out of Range!"}}
+
+factor_en = ["temperature","soc", "charge_rate"]
+factor_de = ["Temperatur","SOC", "Rate der Gebühr"]
 
 lang = "EN"
 
-def battery_is_ok(battery_values):
-	value_range = battery_limits[battery_values["battery_parameter"]]
-	low = value_range[0]
-	high = value_range[1]
-	warning_value = (high * warn_limit) / 100
+def Display_Msg(value, num, attribute):
+	return print(lang_msg[lang][attribute], factor_de[num], ":", value)
+
+def Is_Battery_in_good_condition(condition):
+	if(condition == 1):
+		print(lang_msg[lang]['Okay'])
+		
+def Check_range(value, Num):
+	low = battery_limits[factor_en[Num]]['min']
+	high = battery_limits[factor_en[Num]]['max']
+	warning_value = threshold_val(high)
 	low_warning = low + warning_value
 	high_warning = high - warning_value
-	value = battery_values["value"]
 	if value <= low or value >= high:
-		print(lang_msg[lang]['fail'])
-		return False
-	compare_battery_param_value(low_warning, value)
-	compare_battery_param_value(value, high_warning)
-	return True
-
-def compare_battery_param_value(lower_value, upper_value):
-	if lower_value >= upper_value:
-		print(lang_msg[lang]['warning'])
+		Display_Msg(value, Num, "fail")
+		return 0
+	compare_battery_param_value(low_warning, value, Num, high_warning)
+	return 1
 
+def compare_battery_param_value(lower_value, value, Num, upper_value):
```

Comment:
The name suggests a generic comparison function, but actually it should only be used for warning.
The compulsion of complexity limit has forced you to split the function. But when the smaller functions are not named according to the domain, it makes the code harder to read.
If you can't name according to the domain, maybe another way of splitting is better.